shoop/core/models/addresses.py

```diff
@@ -173,6 +136,46 @@ def as_string_list(self, locale=None):
     def __iter__(self):
         return iter(self.as_string_list())
 
+    def immutable(self):
+        """
+        Get or create saved ImmutableAddress from self.
+
+        :rtype: ImmutableAddress
+        :return: Saved ImmutableAddress with same data as self.
+        """
+        data = get_data_dict(self)
+        return ImmutableAddress.from_data(data)
+
+    def new_mutable(self):
+        """
+        Get a new MutableAddress from self.
+
+        :rtype: MutableAddress
+        :return: Fresh unsaved MutableAddress with same data as self.
+        """
+        data = get_data_dict(self)
+        return MutableAddress.from_data(data)
+
+
+class MutableAddress(Address):
+    """
+    Address that can be changed. Used for `Contact` addresses and for `SavedAddress`.
+    `MutableAddresses` could be created with self.from_data.
+    """
+    pass
+
+
+class ImmutableAddress(ChangeProtected, Address):
+    """
+    Address that cannot be changed. Used for `Order` addresses.
+    `ImmutableAddresses` could be created with self.from_data or by
+    calling self.immutable for existing `MutableAddress`.
+    """
```

Comment:
* Split to title and body.
* Make sure that all references, that are backticked, are actually resolvable as a link by Sphinx, i.e. ``` `ImmutableAddresses` ``` does not work, because there is no class with that name, use ``` `ImmutableAddress` ``` instead.
* Better to just say ``` `ImmutableAddress.from_data` ``` than `self.from_data` to make it clear that it is a class method.
* Also ``` self.immutable for existing `MutableAddress` ``` -> ``` `immutable` method of existing `Address` ```

You can try generating the Sphinx documentation to see how these affect the generated docs. Though the warnings yielded from this will probably be hidden in between other warnings until I get the doc clean-up PR in.